src/commodore-helm/index.ts

```diff
@@ -0,0 +1,243 @@
+        return d.propSource == chartName;
+      });
+      let res: PackageDependency = {
+        depName: d.depName,
+        currentValue: chartVersion,
+      };
+      return res;
+    }
+  );
+
+  return { deps };
+}
+
+export async function extractAllPackageFiles(
+  _config: ExtractConfig,
+  packageFiles: string[]
+): Promise<PackageFile[] | null> {
+  let fileContents: Map<string, string> = new Map();
+  let componentName: string | null = null;
+  let defaults_file: string | null = null;
+
+  for (const file of packageFiles) {
+    const content = await readLocalFile(file, 'utf8');
+    const key = path.parse(file).name;
+    fileContents.set(key, content);
+    if (key != 'defaults') {
+      componentName = key;
+    } else {
+      defaults_file = file;
+    }
+  }
+
+  if (!fileContents.has('defaults')) {
+    logger.error('Component repository has no `class/defaults.ya?ml`');
+    return null;
+  }
+
+  if (!componentName) {
+    logger.error(
+      { packageFiles },
+      'Unable to identify component name from package files'
+    );
+    return null;
+  }
+
+  const defaults: any = yaml.load(fileContents.get('defaults') ?? '');
+  const component: any = yaml.load(fileContents.get(componentName) ?? '');
+
+  if (
+    defaults === undefined ||
+    defaults === null ||
+    defaults.parameters === undefined ||
+    defaults.parameters === null ||
+    component === undefined ||
+    component === null ||
+    component.parameters === undefined ||
+    component.parameters === null
+  ) {
+    return null;
+  }
+
+  const deps: PackageDependency[] = extractHelmChartDependencies(
+    componentName,
+    defaults.parameters,
+    component.parameters
+  );
+
+  if (deps.length == 0) {
+    return null;
+  }
+
+  return [{ packageFile: defaults_file, datasource: 'helm', deps }];
+}
+
+interface KapitanDependency {
+  type: string;
+  source: string;
+  output_path: string;
+}
+interface KapitanHelmDependency extends KapitanDependency {
+  chart_name: string;
+  version: string;
+}
+
+function extractHelmChartDependencies(
+  componentName: string,
+  defaults: Record<string, any>,
+  componentParams: Record<string, any>
+): PackageDependency[] {
+  if (
+    componentName === '' ||
+    defaults === undefined ||
+    defaults === null ||
+    typeof defaults !== 'object' ||
+    componentParams === undefined ||
+    componentParams === null ||
+    typeof componentParams !== 'object'
+  ) {
+    return [];
+  }
+
+  const componentKey: string = componentKeyFromName(componentName);
+
+  const chartVersions: Map<string, string> = defaults[componentKey].charts;
+  if (!chartVersions) {
+    logger.info('No Helm chart versions found');
+    return [];
+  }
+  logger.debug({ chartVersions }, 'chart versions');
+
+  const kapitanHelmDeps: KapitanHelmDependency[] =
+    componentParams.kapitan.dependencies.filter(
+      (dep: KapitanDependency): dep is KapitanHelmDependency =>
+        dep.type === 'helm'
+    );
```

Comment:
I think this also might crash if any of these fields are undefined.

Reply:
Fixed in https://github.com/projectsyn/commodore-renovate/pull/17/commits/94a812cc6599bbac4a97a7b40720795da3a10377